File: backend/routes/trust_evolution.ts
import { Router, Request, Response } from "express";
import { TrustEvolutionEngine } from "../services/trust/trust_evolution_engine";

const router = Router();
const engine = new TrustEvolutionEngine();

router.get("/evolution/:wallet", (req: Request, res: Response) => {
  const wallet = req.params.wallet;
  const growth = engine.calculateTrustGrowth(wallet, "HSP_SETTLEMENT", "att_proof");
  res.json([
    {
      eventId: "ev_init",
      timestamp: new Date().toISOString(),
      previousScore: growth.previousScore,
      newScore: growth.newScore,
      change: growth.change,
      reason: growth.reason
    }
  ]);
});

router.get("/before-after/:wallet", (req: Request, res: Response) => {
  const wallet = req.params.wallet;
  const snapshot = engine.generateBeforeAfterSnapshot(wallet);
  res.json(snapshot);
});

router.get("/next-level/:wallet", (req: Request, res: Response) => {
  const wallet = req.params.wallet;
  const milestone = engine.calculateNextMilestone(wallet);
  res.json(milestone);
});

export default router;

File: backend/app/api/v1/trust_evolution.py
from fastapi import APIRouter, HTTPException, Path
from app.services.trust.trust_evolution_engine import TrustEvolutionEngine
from app.database.reputation_history import get_reputation_history

router = APIRouter(
    prefix="/trust",
    tags=["Trust Evolution Flywheel Engine"]
)

@router.get("/evolution/{wallet}")
def get_evolution_timeline(wallet: str = Path(..., description="Wallet address")):
    try:
        history = get_reputation_history(wallet)
        timeline = []
        for i, ev in enumerate(history.get("events", [])):
            timeline.append({
                "eventId": f"ev_{i}",
                "timestamp": ev.get("timestamp"),
                "previousScore": ev.get("previous_score", 300),
                "newScore": ev.get("current_score", 300),
                "change": f"+{ev.get('delta', 0)}" if ev.get("delta", 0) >= 0 else str(ev.get("delta", 0)),
                "reason": ev.get("reason", "Verified transaction registered")
            })
        return timeline
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/before-after/{wallet}")
def get_before_after(wallet: str = Path(..., description="Wallet address")):
    try:
        engine = TrustEvolutionEngine()
        return engine.generateBeforeAfterSnapshot(wallet)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/next-level/{wallet}")
def get_next_level(wallet: str = Path(..., description="Wallet address")):
    try:
        engine = TrustEvolutionEngine()
        return engine.calculateNextMilestone(wallet)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
